prom_wrapper.coffee

```diff
@@ -0,0 +1,114 @@
+prom = require('prom-client')
+registry = require('prom-client').register
+metrics = {}
+
+
+optsKey = (opts) ->
+	keys = Object.keys(opts)
+	return '' if keys.length == 0
+
+	keys = keys.sort()
+
+	hash = '';
+	for key in keys
+		hash += "," if hash.length
+		hash += "#{key}:#{opts[key]}"
+
+	return hash
+
+extendOpts = (opts, labelNames) ->
+	for label in labelNames
+		opts[label] ||= ''
+	return opts
+
+optsAsArgs = (opts, labelNames) ->
+	args = []
+	for label in labelNames
+		args.push(opts[label] || '')
+	return args
+
+
+PromWrapper =
+	ttlInMinutes: 0
+	registry: registry
+
+	metric: (type, name) ->
+		registry.getSingleMetric(name) || new MetricWrapper(type, name)
+
+	collectDefaultMetrics: prom.collectDefaultMetrics
+
+
+class MetricWrapper
+	constructor: (type, name) ->
+		metrics[name] = this
+		@name = name
+		@instances = {}
+		@lastAccess = new Date()
+		@metric = switch type
+			when "counter"
+				new prom.Counter({
+					name: name,
+					help: name,
+					labelNames: ['app','host','status','method', 'path']
+				})
+			when "summary"
+				new prom.Summary({
+					name: name,
+					help: name,
+					maxAgeSeconds: 600,
+					ageBuckets: 10,
+					labelNames: ['app', 'host', 'path', 'status_code', 'method', 'collection', 'query']
+				})
+			when "gauge"
+				prom.Gauge({
+					name: name,
+					help: name,
+					labelNames: ['app','host', 'status']
+				})
+
+	inc: (opts, value) ->
+		@_execMethod 'inc', opts, value
+
+	observe: (opts, value) ->
+		@_execMethod 'observe', opts, value
+
+	set: (opts, value) ->
+		@_execMethod 'set', opts, value
+
+	sweep: () ->
+		thresh = new Date(Date.now() - 1000 * 60 * PromWrapper.ttlInMinutes)
+		for key in Object.keys(@instances)
```

Comment:
PromWrapper.ttlInMinutes is set to 0 by default which would set the threshold to 'now' and so try and expire most metrics? Worth giving this a little bit of leeway, say with a 60 second ttl?

Reply:
The timer method won't call `sweep()` if `ttlInMinutes` is zero, so the default behaviour is to do the same as before.
https://github.com/overleaf/metrics-module/blob/e3079c44e7720f19cb33a5d98ffa0095aa1c6e06/prom_wrapper.coffee#L106